plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/TestSqlServerConnectorTest.java

```diff
@@ -151,6 +155,22 @@ public void testInsertWriteBulkinessWithTimestamps(String timestampType)
         }
     }
 
+    @Test(dataProvider = "testTableNameDataProvider")
+    public void testCreateTableWithSpecialCharacterName(String tableName)
+    {
+        if (!hasBehavior(SUPPORTS_CREATE_TABLE)) {
+            assertQueryFails("CREATE TABLE " + "\"" + tableName + "\"" + " (a bigint, b double, c varchar(50))", "This connector does not support creating tables");
+            return;
+        }
+
+        assertUpdate("CREATE TABLE " + "\"" + tableName + "\"" + " (a bigint, b double, c varchar(50))");
+        assertTrue(getQueryRunner().tableExists(getSession(), tableName.replace("\"\"", "\"")));
+        assertTableColumnNames("\"" + tableName + "\"", "a", "b", "c");
+
+        assertUpdate("DROP TABLE " + "\"" + tableName + "\"");
+        assertFalse(getQueryRunner().tableExists(getSession(), tableName));
```

Comment:
`tableExists` call here should be identitcal to the one above, otherwise it looks like you're testing with a wrong name here, and not really checking whether the DROP was effective

 (but i think this line is correct, actually)

Reply:
Yes the problem is that when we execute : 
` assertUpdate("DROP TABLE " + "\"" + tableName + "\"");`
this actualy going as trino cli request and it has some rules over passing special characters so you need to pass it in quotes, or you will get:
`trino:dbo> drop table te[ st3;`
`Query 20221001_092755_00035_jm82s failed: line 1:14: mismatched input '['. Expecting: '.', <EOF>`
`drop table te[ st3 `

But when we execute smth like this : 
`assertFalse(getQueryRunner().tableExists(getSession(), tableName));`

You query metadata from db without the same trino parsing rules, you need just strict literal name

Reply:
okay, so this is same problem as in https://github.com/trinodb/trino/pull/14286#discussion_r985546687